src/eddymotion/model/tests/test_utils.py

```diff
@@ -0,0 +1,30 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2024 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+import numpy as np
+from eddymotion.model.utils import (
+    calculate_angle_matrix,
+)
+
+
+def test_calculate_angle_matrix():
```

Comment:
I tested the angle computations while working on the hyperparameter optimization. I'm not sure if this aligns with the tests you wanted to do, but I''m dropping my code it here in case it's useful (feel free to ignore if it's not).
```
def test_calculate_angle_matrix():
    # Define test gradient vectors
    gradients = np.array([
        [1, 0, 0],
        [0, 1, 0],
        [0, 0, 1]
    ])
    
    # Calculate the angle matrix
    angle_matrix = calculate_angle_matrix(gradients)
    
    # Expected angle matrix
    expected_angles = np.array([
        [0, np.pi / 2, np.pi / 2],
        [np.pi / 2, 0, np.pi / 2],
        [np.pi / 2, np.pi / 2, 0]
    ])
    
    # Assert the shape of the angle matrix
    assert angle_matrix.shape == expected_angles.shape
    
    # Assert the calculated angles are almost equal to the expected angles
    np.testing.assert_array_almost_equal(angle_matrix, expected_angles, decimal=6)

if __name__ == "__main__":
    pytest.main()
```

Reply:
Thanks Elodie. I add that to my last push force, and incorporated other cases for the array to be rectangular.